### ISSUE TITLE
EmptyLineSeparator: false negative for 'import module' (MODULE_IMPORT) followed by a type declaration

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20111]
Parent Issue https://github.com/checkstyle/checkstyle/issues/19966

I have read check documentation: https://checkstyle.org/checks/whitespace/emptylineseparator.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtd/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyLineSeparator"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputModuleImport.java <<'EOF'
import module java.base;
public class InputModuleImport {
}
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputRegularImport.java <<'EOF'
import java.util.List;
public class InputRegularImport {
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar checkstyle-13.5.0-all.jar -c config.xml InputModuleImport.java

Starting audit...
Audit done.

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar checkstyle-13.5.0-all.jar -c config.xml InputRegularImport.java

Starting audit...
[ERROR] /Users/vivek/checkstyle/InputRegularImport.java:2:1: 'CLASS_DEF' should be separated from previous line. [EmptyLineSeparator]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

Both files have the same layout: an import  followed by a class declaration  with no blank line between them. The regular `import` case correctly reports the missing blank line, but the `import module` (`MODULE_IMPORT`) case produces no violation. The module-import file should report the same way:

`InputModuleImport.java:2:1: 'CLASS_DEF' should be separated from previous line. [EmptyLineSeparator]`
